src/GuessTheNumber.java

```diff
@@ -1,21 +1,29 @@
 public class GuessTheNumber {
+    private int tries;
     private final int number;
     private boolean isOver = false;
+    private IOInterface io;
+    int input;
```

Comment:
All the fields are private, but input is not. Was this on purpose? Might be good to be consistent with the visibility of the variables.